README.md

```diff
@@ -211,6 +219,103 @@ import {configure, Lang} from '@gravity-ui/page-constructor';
 configure({lang: Lang.En});
 ```
 
+### Analytics
+
+#### Init
+
+To start using any analytics, pass a handler to the constructor. The handler must be created on a project side. The handler will receive the `default` and `custom` event objects. The passed handler will be fired on a button, link, navigation, and control clicks. As one handler is used for all events treatment, pay attention to how to treat different events while creating the handler. There are predefined fields that serve to help you to build complex logic.
+
+Pass `autoEvents: true` to constructor to fire automatically configured events.
+
+```ts
+function sendEvents(events: MyEventType []) {
+  ...
+}
+
+<PageConstructorProvider
+    ...
+
+    analytics={{sendEvents, autoEvents: true}}
+
+    ...
+/>
+```
+
+An event object has only one required field - `name`. It also has predefined fields, which serve to help manage complex logic. For example, `counter.include` can help to send event in a particular counter if several analytics systems are used in a project.
+
+```ts
+type AnalyticsCounters = {
+  include?: string[];
+  exclude?: string[];
+};
+
+type AnalyticsEvent<T = {}> = T & {
+  name: string;
+  type?: string;
+  counters?: AnalyticsCounters;
+  blockName?: string;
+};
+```
+
+It is possible to configure an event type needed for a project.
+
+```ts
+type MyEventType = AnalyticsEvent<{
+  anyParameterName?: string; // only a 'string' type is supported
+}>;
+```
+
+#### Counter selector
+
+It is possible to configure an event to which an analytics system to sent.
+
+```ts
+type AnalyticsCounters = {
+  include?: string[];
+  exclude?: string[];
+};
+```
+
+#### blockName parameter
+
+Pass `blockName` value to define place on a project where an event is fired.
+
+Use selector below or create logic that serves project needs.
+
+```ts
+// utils.ts
+const isCounterAllowed = (counter: Counter, counters?: AnalyticsCounters) => {
+  if (!counters) {
+    return true;
+  }
+
+  if (counters.exclude?.includes(counter)) {
+    return false;
+  } else if (counters.include?.includes(counter)) {
+    return true;
+  }
+
+  return false;
+};
+
+// analyticsHandler.ts
+if (isCounterAllowed(counterName, counters)) {
+  analyticsCounter.reachGoal(counterName, name, parameters);
+}
+```
+
+#### Reserved event types
+
+Several predefined event types are used to mark automatically configured events. Use the types to filter default events, for example.
+
+```ts
+enum PredefinedEventTypes {
+  Default = 'default-event', // default events which fire on every button click
```

Comment:
is it 'button-click' event mb?

Reply:
most of the events fired in `page-constructor` are `button-clicks`
the meaning of this item is to highlight that fired event is not setup  by users but is predefined by package itself
I think `default-event` is good to describe its meaning. 
It  may be called `automatic-event` or `auto-click` (something like that. But I do not like them)